src/java/org/apache/cassandra/index/sai/disk/format/Version.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.cassandra.index.sai.disk.format;
+
+import java.util.List;
+
+import com.google.common.base.Objects;
+import com.google.common.collect.Lists;
+
+import org.apache.cassandra.index.sai.IndexContext;
+import org.apache.cassandra.index.sai.disk.v1.V1OnDiskFormat;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+/**
+ * Format version of indexing component, denoted as [major][minor]. Same forward-compatibility rules apply as to
+ * {@link org.apache.cassandra.io.sstable.format.Version}.
+ */
+public class Version
+{
+    // Current version
+    public static final Version BA = new Version("ba", V1OnDiskFormat.instance, (c, i) -> defaultFileNameFormat(c, i, "ba"));
```

Comment:
Shouldn't it start at `aa`?

Reply:
Was there a DataStax internal `aa`? :)

Reply:
There was an internal version `aa` and, we had decided that it might make sense to start the versions here from `ba` which matches the current internal row-aware version. As such, it might make more sense to change this to `aa` and handle this, if and when, internally.